keys/generate-key-pair.sh

```diff
@@ -54,39 +55,73 @@ function allKeys() {
   for i in "rsa" "ecdsa256" "ecdsa384"; do
     keyType=$i
     genKey
-    wait
   done
 }
 
 #This function will create a private key that is needed to create a private keystore. Encryption keyType passed will decide which command to run for private key creation
 function genKey() {
+  local privateKey=""${keyType}"private-key.pem"
+  local keyCert=""${keyType}"Cert.crt"
   #Check if the folder is already created for the keyType (In case of multiple runs)
   mkdir -p "${keyType}"Key && pushd "${keyType}"Key >/dev/null || return
   #Generate a private RSA key.
   if [[ $keyType == "rsa" ]]; then
-    echo -e "Generating a "${keyType}" private key."
-    openssl genrsa -out "${keyType}"private-key.pem 2048 >/var/tmp/out 2>&1
-    chk $? 'Generating a rsa private key.'
+    if [ "$(uname)" == "Darwin" ]; then
+      echo "Using macOS, will generate private key and certificate simultaneously."
+      gencmd="openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout "$privateKey" -out "$keyCert""
```

Comment:
FYI: when you put a command in a variable like this, you have to take care to preserve quotes around arguments that might need them, for example file names. The way you have it now, if `$privateKey` or `$keyCert` had spaces in the file name, the cmd would fail. In this case, your script controls the value of those variables, so you know there are no spaces, so it is not a problem.

In the future, it is safer to do:

```
gencmd="openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout '$privateKey' -out '$keyCert' "
```

That way, the double quotes around the whole thing mean those 2 variables will get evaluated when putting the cmd string into `gencmd`. The when gencmd is executed, the single quotes will be around the literal file names, holding them together if they have any spaces in them.